src/client/components/app/LineDetailsDisplay/PositionsTable.tsx

```diff
@@ -232,7 +242,127 @@ export const PositionsTable = ({ positions, displayLine = false }: PositionsProp
     return [];
   };
 
-  console.log('positions table display line', displayLine, !displayLine);
+  // Returns a list of transactions to display on positions table for proposals
+  const getUserProposalActions = (proposal: CreditProposal) => {
+    const approveMutualConsent = {
+      name: t('components.transaction.add-credit.accept-terms'),
+      handler: depositHandler,
+      disabled: false,
+    };
+
+    const revokeMutualConsent = {
+      name: t('components.transaction.revoke-consent.cta'),
+      handler: revokeConsentHandler,
+      disabled: false,
+    };
+
+    // Display button to approve proposal for the borrower of the proposal
+    if (userRoleMetadata.role === BORROWER_POSITION_ROLE) {
```

Comment:
should be `!proposal.make` right not hardcoded Borrower? can add `else` branch to `if(proposal.maker)` in line below

Reply:
updated logic to use `borrower` address from `selectedLine` and compare to the `userWallet` from the `state.wallet`. 

if user wallet is `line.borrower` and `proposal.maker` the user will see both `approveConsent` and `revokeConsent`

otherwise, user will see either `approveMutualConsent` or `revokeMutualConsent` based on whether the `userWallet` is equivalent to `line.borrower` or `proposal.maker`